packages/angular/cli/utilities/package-manager.ts

```diff
@@ -6,55 +6,87 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import { execSync } from 'child_process';
-import { existsSync } from 'fs';
+import { exec, execSync } from 'child_process';
+import { constants, promises as fs } from 'fs';
 import { join } from 'path';
 import { satisfies, valid } from 'semver';
+import { promisify } from 'util';
 import { PackageManager } from '../lib/config/workspace-schema';
 import { getConfiguredPackageManager } from './config';
 
-function supports(name: string): boolean {
+const execPromise = promisify(exec);
+async function supports(name: PackageManager): Promise<boolean> {
   try {
-    execSync(`${name} --version`, { stdio: 'ignore' });
+    await execPromise(`${name} --version`);
```

Comment:
Nit: I suggest naming the variable you want to use the most intuitive name `exec`, and name the other one with the qualifier `execCallback`, that way future changes to this file will naturally use the correct symbol.

The counterpoint for this particular case is that `exec` is a fairly well-known Node API, so having that return a `Promise` can be counter-intuitive for some. Ultimately up to you, just a suggestion.

Reply:
Let's rename it to `exec`